main/src/core/MonoDevelop.Core/MonoDevelop.Core/FileService.cs

```diff
@@ -169,6 +186,33 @@ public static void DeleteDirectory (string path)
 			OnFileRemoved (new FileEventArgs (path, true));
 		}
 
+		/// <summary>
+		/// Checks if a directory can be safely deleted. It checks if the directory is not a system relevant directory.
+		/// </summary>
+		/// <param name="path">The path to be checked.</param>
+		/// <param name="requiredParentDirectory">optional parameter that specifies a required parent of the path.</param>
+		/// <exception cref="InvalidOperationException">Is thrown when the directory can't be safely deleted.</exception>
+		public static void AssertCanDeleteDirectory (FilePath path, string requiredParentDirectory = null, bool canDeleteParent = true)
```

Comment:
Pls add the new param to the xml comment and maybe use a better name? `canDeleteParent` is confusing, what am I checking? Parent or path?
```suggestion
		public static void AssertCanDeleteDirectory (FilePath path, string requiredParentDirectory = null, bool includingParent = true)
```

Reply:
added docs. I don't think the old name is confusing because it basically says what it is for.